airbyte-integrations/connectors/source-google-ads/source_google_ads/streams.py

```diff
@@ -94,8 +95,10 @@ def parse_response(self, response: SearchPager) -> Iterable[Mapping]:
             yield self.google_ads_client.parse_single_result(self.get_json_schema(), result)
 
     def read_records(self, sync_mode, stream_slice: Mapping[str, Any] = None, **kwargs) -> Iterable[Mapping[str, Any]]:
-        response = self.google_ads_client.send_request(self.get_query(stream_slice))
-        yield from self.parse_response(response)
+        stream_slice = stream_slice or {}
+        account_responses = self.google_ads_client.send_request(self.get_query(stream_slice), customer_id=self._customer_id)
```

Comment:
according to your logic customer_id will be None

Reply:
I updated the code a bit, overridden the `stream_slices` method for the class responsible for `Full Refresh`. Unfortunately, the solution is not very beautiful, but it allows you to avoid redefining the `_read_incremental` method for the SourceGoogleAds class, and at the same time be able to get the current `customer_id` from any class method (in particular, for the `get_updated_state` method).